samza-core/src/main/java/org/apache/samza/task/StreamOperatorTask.java

```diff
@@ -146,6 +146,11 @@ public final void processAsync(IncomingMessageEnvelope ime, MessageCollector col
               callback.complete();
             }
           });
+        } else {
+          // If InputOperator is not found in the operator graph for a given SystemStream, throw an exception else the
+          // job will timeout due to async task callback timeout (TaskCallbackTimeoutException)
+          callback.failure(new SamzaException(
+              String.format("InputOperator not found in OperatorGraph for %s", systemStream.toString())));
```

Comment:
nit: `systemStream.toString()` seems redundant. passing `systemStream` should be sufficient.